Lib/gftools/builder/ninja.py

```diff
@@ -0,0 +1,289 @@
+            self.w.rule("variable", "fontmake -o variable -m $in $fontmake_args")
+
+        self.w.comment("Build a set of instance UFOs from Designspace")
+        self.w.rule("instanceufo", "fontmake -i -o ufo -m $in $fontmake_args")
+
+        self.w.comment("Build a TTF file from a UFO")
+        self.w.rule(
+            "buildttf", "fontmake -o ttf -u $in $fontmake_args --output-path $out"
+        )
+
+        self.w.comment("Build an OTF file from a UFO")
+        self.w.rule(
+            "buildotf", "fontmake -o otf -u $in $fontmake_args --output-path $out"
+        )
+
+        self.w.comment("Add a STAT table to a set of variable fonts")
+        self.w.rule(
+            "genstat",
+            "gftools-gen-stat.py --inplace $other_args --axis-order $axis_order -- $in  && touch $stampfile",
+        )
+
+        self.w.comment("Run the font fixer in-place and touch a stamp file")
+        self.w.rule(
+            "fix", "gftools-fix-font.py -o $in $fixargs $in && touch $in.fixstamp"
+        )
+
+        self.w.comment("Run the ttfautohint in-place and touch a stamp file")
+        self.w.rule(
+            "autohint",
+            "ttfautohint $in $in.autohinted && mv $in.autohinted $in && touch $in.autohintstamp",
+        )
+
+        self.w.comment("Create a web font")
+        self.w.rule("webfont", f"fonttools ttLib.woff2 compress -o $out $in")
+
+        self.w.newline()
+
+    def get_designspaces(self):
+        self.designspaces = []
+        for source in self.config["sources"]:
+            if source.endswith(".glyphs"):
+                builder = UFOBuilder(glyphsLib.GSFont(source))
+                # This is a sneaky way of skipping the hard work of
+                # converting all the glyphs and stuff, and just gettting
+                # a minimal designspace
+                builder.to_ufo_groups = (
+                    builder.to_ufo_kerning
+                ) = builder.to_ufo_layers = lambda: True
+
+                designspace = builder.designspace
+                designspace_path = os.path.join("master_ufo", designspace.filename)
+                os.makedirs(os.path.dirname(designspace_path), exist_ok=True)
+                designspace.write(designspace_path)
+                self.w.comment("Convert glyphs source to designspace")
+                designspace_and_ufos = [designspace_path] + list(
+                    set(
+                        [
+                            os.path.join("master_ufo", m.filename)
+                            for m in designspace.sources
+                        ]
+                    )
+                )
+                self.w.build(designspace_and_ufos, "glyphs2ufo", source)
+            else:
+                designspace_path = source
+                designspace = DesignSpaceDocument.fromfile(designspace_path)
+            self.designspaces.append((designspace_path, designspace))
+        self.w.newline()
+
+    def fontmake_args(self, args):
```

Comment:
Is this function where we could add the additional argument to `--expand-features-to-instances`? 

See https://github.com/googlefonts/gftools/issues/581 for more background on this.

Reply:
Yes, that's where it should go.

Reply:
Sorry, I only intended to highlight Line 131 here.

I tried hacking this in the download within my local venv, but at least in initial tests, it wasn't working as I had hoped.

Reply:
Hmm, yeah, I’ve edited this in my venv, but I can’t even get it to print debugging statements (even using a `config.yaml` file with `logLevel` set to `DEBUG`. I’m not entirely sure if the Ninja builder is even running, or even how I would determine that.

```python
    def fontmake_args(self, args):
        my_args = []
        my_args.append("--expand-features-to-instances") # TESTING;
        my_args.append("--filter ...")
        if self.config["flattenComponents"]:
            my_args.append("--filter FlattenComponentsFilter")
        if self.config["decomposeTransformedComponents"]:
            my_args.append("--filter DecomposeTransformedComponentsFilter")
        if "output_dir" in args:
            my_args.append("--output-dir " + args["output_dir"])
        if "output_path" in args:
            my_args.append("--output-path " + args["output_path"])

        print(" ".join(my_args)) # DEBUGGING;
        
        return " ".join(my_args)
```

Reply:
Thank you for the tip! I’ll give that a try.

Reply:
Awesome, that definitely ran the Ninja builder, including my hacks!

However (this may be stuff you already know), it seems that:
- it will only run on a config file, not a designspace
- add the arg as suggested yields the following part way through the build: `fontmake: error: "--expand-features-to-instances" option invalid for UFO source`
- It expects/requires stat table info in the config

```
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/stephennixon/type-repos/shantell-sans/venv-ninja/lib/python3.10/site-packages/gftools/builder/ninja.py", line 289, in <module>
    NinjaBuilder(sys.argv[1]).build()
  File "/Users/stephennixon/type-repos/shantell-sans/venv-ninja/lib/python3.10/site-packages/gftools/builder/ninja.py", line 47, in build
    os.remove(temporary)
FileNotFoundError: [Errno 2] No such file or directory: './fonts/builder/variable/shantell_sans-ital_wght_BNCE_IRGL_TRAK--static-ninja[wght].ttf.statstamp'
```

Reply:
1) Yeah, that's true when you run it as `python3 -m gftools.builder.ninja`, but it's meant to work as `gftools-builder` as well.
2) Urgh. We need to add that only when creating UFOs.
3) No, that's a problem removing temporary files, which I think I fixed.

Reply:
Got it. Thanks for the insights here! I might try to hack away at this on my end, but don't let my issues block an otherwise solid PR.